includes/tuque_wrapper.inc

```diff
@@ -233,6 +233,56 @@ class IslandoraFedoraObject extends FedoraObject {
       throw $e;
     }
   }
+
+  /**
+   * Purge a datastream.
+   *
+   * Invokes datastream altered/purged hooks and calls the API-M method.
+   *
+   * @see FedoraObject::purgeObject()
+   */
+  public function purgeDatastream($id) {
+    $this->populateDatastreams();
+
+    if (!array_key_exists($id, $this->datastreams)) {
+      return FALSE;
+    }
+    $context = array(
+      'action' => 'purge',
+      'purge' => TRUE,
+      'delete' => FALSE,
+      'block' => FALSE,
+    );
+    islandora_alter_datastream($this, $this[$id], $context);
```

Comment:
Alter should probably be moved inside the `try` block below (for more consistent exception handling/reporting, in case something _exceptional_ occurs).